internal/middlewares/auth.go

```diff
@@ -0,0 +1,75 @@
+package middlewares
+
+import (
+	"errors"
+	"log"
+	"net/http"
+
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+const authErrorHeader = "Authentication error: "
+
+type serviceRequest struct {
+	clientID string
+	account  string
+	psk      string
+}
+
+func newServiceRequest(clientID, account, psk string) (*serviceRequest, error) {
+	switch {
+	case clientID == "":
+		return nil, errors.New(authErrorHeader + "Missing receptor-controller-client-id header")
+	case account == "":
+		return nil, errors.New(authErrorHeader + "Missing receptor-controller-account header")
+	case psk == "":
+		return nil, errors.New(authErrorHeader + "Missing receptor-controller-psk header")
+	}
+	return &serviceRequest{
+		clientID: clientID,
+		account:  account,
+		psk:      psk,
+	}, nil
+}
+
+func (sr *serviceRequest) validate(secrets map[string]interface{}) error {
```

Comment:
Seems like maybe the validate method needs to be moved to a SeriviceCredentialsValidator object.  It would be that objects responsibility to validate the server-to-service credentials.

Reply:
:+1: